subcommand/webhook-cert-manager/command.go

```diff
@@ -0,0 +1,277 @@
+			// Bundle is updated, set it up
+
+		case <-time.After(30 * time.Minute):
+			// This forces the mutating ctrlWebhook config to remain updated
+			// fairly quickly. This is done every 30 minutes to ensure the certificates
+			// are in sync. Because the certificate and key are being read from a secret,
+			// this does not have to be processed as aggressively as the 1 sec time in
+			// the connect inject cert watcher.
+
+		case <-ctx.Done():
+			// Quit
+			return
+		}
+
+		log.Info("getting secret from kubernetes")
+		certSecret, err := clientset.CoreV1().Secrets(bundle.SecretNamespace).Get(ctx, bundle.SecretName, metav1.GetOptions{})
+		if err != nil && k8serrors.IsNotFound(err) {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      bundle.SecretName,
+					Namespace: bundle.SecretNamespace,
+				},
+				Data: map[string][]byte{
+					corev1.TLSCertKey:       bundle.Cert,
+					corev1.TLSPrivateKeyKey: bundle.Key,
+				},
+				Type: corev1.SecretTypeTLS,
+			}
+
+			log.Info("creating kubernetes secret")
+			if _, err := clientset.CoreV1().Secrets(bundle.SecretNamespace).Create(ctx, secret, metav1.CreateOptions{}); err != nil {
+				log.Error(fmt.Sprintf("error writing secret to API server: %s", err))
+				continue
+			}
+
+			log.Info("updating webhook configuration with new CA")
+			if err := c.updateWebhookConfig(ctx, bundle, clientset); err != nil {
+				log.Error(fmt.Sprintf("error updating webhook configuration: %s", err))
+				continue
+			}
+			continue
+		} else if err != nil {
+			log.Error(fmt.Sprintf("error getting secret for API server: %s", err))
+			continue
+		}
+
+		// Don't update secret if the certificate is unchanged.
+		if bytes.Equal(certSecret.Data[corev1.TLSCertKey], bundle.Cert) {
+			continue
+		}
+
+		certSecret.Data[corev1.TLSCertKey] = bundle.Cert
+		certSecret.Data[corev1.TLSPrivateKeyKey] = bundle.Key
+
+		log.Info("updating secret with new certificate")
+		_, err = clientset.CoreV1().Secrets(bundle.SecretNamespace).Update(ctx, certSecret, metav1.UpdateOptions{})
+		if err != nil {
+			log.Error(fmt.Sprintf("error updating secret with certificate: %s", err))
+			continue
+		}
+
+		log.Info("updating webhook configuration with new CA")
+		if err := c.updateWebhookConfig(ctx, bundle, clientset); err != nil {
+			log.Error(fmt.Sprintf("error updating webhook configuration: %s", err))
+			continue
+		}
+	}
+}
+
+func (c *Command) updateWebhookConfig(ctx context.Context, metaBundle cert.MetaBundle, clientset kubernetes.Interface) error {
+	if len(metaBundle.CACert) > 0 {
+		value := base64.StdEncoding.EncodeToString(metaBundle.CACert)
+
+		webhookCfg, err := clientset.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(ctx, metaBundle.WebhookConfigName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		var patches []patch
+		for i := range webhookCfg.Webhooks {
+			patches = append(patches, patch{
```

Comment:
should we check here the name of the `i`-th webhook and only update it if it is the webhook for which we're updating the CA?

Reply:
We will be updating the CA on every single webhook defined in the webhook configuration as they will share the same CA and cert.